API.md

```diff
@@ -1,39 +1,293 @@
 # API Reference <a name="API Reference" id="api-reference"></a>
 
+## Constructs <a name="Constructs" id="Constructs"></a>
```

Comment:
This is rad...does it get auto-generated? (and if so, how)

Reply:
Yup, autogenerated from the code comments using docgen. Check out `projen docgen` for details.